annotation-processor/src/main/java/com/hubspot/annotation/processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2014-2018 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hubspot.annotation.processor;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic.Kind;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import com.google.auto.service.AutoService;
+
+@SupportedAnnotationTypes("java.lang.SuppressWarnings")
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@AutoService(Processor.class)
+public class ModernizerAnnotationProcessor extends AbstractProcessor {
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        List<String> annotatedClasses;
+        for (TypeElement annotation : annotations) {
+            annotatedClasses = getAnnotatedClasses(roundEnv, annotation);
+            makeAnnotatedClassesFile(annotatedClasses);
+        }
+        return true;
+    }
+
+    private List<String> getAnnotatedClasses(
+        RoundEnvironment roundEnv, TypeElement annotation) {
+        List<String> annotatedClasses = new ArrayList<>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings =
+                Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        FileObject fileObject;
+        Writer writer;
+        try {
+            fileObject = processingEnv.getFiler().createResource(
+                StandardLocation.CLASS_OUTPUT, "",
```

Comment:
I realized that this will cause (I think) the generated files to be included in the compiled jar. Maybe putting this in `SOURCE_OUTPUT` makes more sense? In any case, we should investigate this to avoid adding generated files as a resource to any porject that this is run on.

Reply:
We are creating the file in the annotation processing stage. Probably we can just delete the file after reading it in the plugin mojo. Not sure if this is a good approach though.

Reply:
It would be best to find a way to exclude it from the classpath of the artifact without just deleting it. We can experiment.

Reply:
I think we want to put this in the `target` directory, but not `target/classes` (which is where Maven will point `CLASS_OUTPUT`). It's probably poor form, but could we just manually create the file ourselves and ignore the `processingEnv` entirely?

Reply:
We want to create this file in the processing environment of the processor right?

Reply:
Created the file under `target/modernizer` dir. This is assuming that the outputDirectory is the default one (since it covers most of the cases). Should figure out what needs to be done in other cases.